evm/p2p/kademlia.py

```diff
@@ -182,49 +191,50 @@ def add(self, node):
         else:
             self.replacement_cache.append(node)
             return self.head
+        return None
 
     @property
-    def head(self):
+    def head(self) -> Node:
         """Least recently seen"""
         return self.nodes[0]
 
-    def __contains__(self, node):
+    def __contains__(self, node: Node) -> bool:
         return node in self.nodes
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self.nodes)
 
-    def __lt__(self, other):
+    def __lt__(self, other: 'KBucket') -> bool:
         if not isinstance(other, self.__class__):
             raise TypeError("Cannot compare KBucket with type {}.".format(other.__class__))
         return self.end < other.start
 
 
 class RoutingTable:
```

Comment:
For consistency, do you want to add a `collections.abc` base class to this as well?

Reply:
I wanted to use `collections.abc.Collection` but that was only added in 3.6, so used `Iterable` and  `Container`